### ISSUE TITLE
help command

### ISSUE THREAD

**Issue by staabm** [clue/phar-composer#41]
`php phar-composer.phar` directly starts the app in an interactive mode.

Usually CLI binaries output a help page when opened without any args, not sure this would make sense for phar-composer though.

One thing which bugs me even more is that `php phar-composer.phar help` does not contain any output about available commands. ATM it looks like the default symfony cli args, but nothing phar-composer related:

```
$ php phar-composer.phar help
Usage:
 help [--xml] [--format="..."] [--raw] [command_name]

Arguments:
 command               The command to execute
 command_name          The command name (default: "help")

Options:
 --xml                 To output help as XML
 --format              To output help in other formats (default: "txt")
 --raw                 To output raw command help
 --help (-h)           Display this help message.
 --quiet (-q)          Do not output any message.
 --verbose (-v|vv|vvv) Increase the verbosity of messages: 1 for normal output,
2 for more verbose output and 3 for debug
 --version (-V)        Display this application version.
 --ansi                Force ANSI output.
 --no-ansi             Disable ANSI output.
 --no-interaction (-n) Do not ask any interactive question.

Help:
 The help command displays help for a given command:

   php phar-composer.phar help list

 You can also output the help in other formats by using the --format option:

   php phar-composer.phar help --format=xml list

 To display the list of available commands, please use the list command.
```

**Comment by ppetermann** [comment]
use list instead of help, its a lot more helpful ;)

**Comment by clue** [comment]
> use list instead of help, its a lot more helpful ;)

Agreed – although I agree it's a bit strange that there's a command that provides more help than the `help` command :)

> I am not aware of a cli component which does a better job than the symfony one.

Too bad… I'm a bit curious whether this also is an issue for other/similar tools?

> I think it could be worth to add a "type help for information about usage" as a welcome message 

Not sure which "welcome message" you're referring to, can you elaborate?

At the very least, we should probably add a section to our README, I suppose?

**Comment by ppetermann** [comment]
there is none - right now you get an input prompt and nothing - telling people how to get help should be the minimum.

personally I'd prefer it like most tools are doing it - show help/list when no command is given

**Comment by clue** [comment]
Indeed, this dialog could probably be a bit more user friendly :-)

```bash
$ php bin/phar-composer 
Enter (partial) project name > …
```

Thanks for your suggestion! :+1: Anybody willing to file a PR? :-)

**Comment by staabm** [comment]
> Too bad… I'm a bit curious whether this also is an issue for other/similar tools?

Recently I stumbled over climate, which also looks promising (and less overengineered like sfConsole)
https://github.com/thephpleague/climate

**Comment by clue** [comment]
Closed as out of scope as per #88
